### PR TITLE
Blacken

### LINKED ISSUE #131: Black-ening?
@martindurant Would you consider adopting [black](https://black.readthedocs.io/en/stable/index.html) as the code format for this repo? I've just noticed in my two recent PRs that the "diff radius" was a bit larger than I'd intended due to accidental code formatting in my editor. Having a pre-established code format may be useful if more folks begin joining this project.

I would be happy to open a PR for this. It would include:

- Setup up a pre-commit config for code formatting.
- Add a code formatting check to your existing CI.
- Blacken the code base.

### PR DESCRIPTION
**Pull request by asford**
Fixes #131.

Add a basic pre-commit configuration invoking black, and adds a pre-commit based check to the Travis build.

Blacken the full code base via `pre-commit run --all`.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -45,6 +45,19 @@ matrix:
       script:
         - py.test -vv s3fs
 
+    - language: generic
+
+      env:
+        - TEST=PRE-COMMIT
+
+      install:
+        - pip install pre-commit
```
Comment (martindurant) [inline .travis.yml]: If the language is generic, you need to install python - this command is trying to install black into the system python. I would not be against simply adding "black" as the final step of the main tests, which is what dask does.

File: README.md [hunk 2]
```diff
@@ -28,3 +28,36 @@ mounting of the file-system implementation may be available for all implementati
...
+## Develop
+
+### Code Formatting
+
+Filesystem_spec uses [Black](https://black.readthedocs.io/en/stable) to ensure
+a consistent code format throughout the project. ``black`` can be installed
+with:
+
+```
+pip install black
```
Comment (martindurant) [inline README.md]: (or conda!)

File: README.md [hunk 3]
```diff
@@ -28,3 +28,36 @@ mounting of the file-system implementation may be available for all implementati
...
+
+to auto-format your code. Additionally, many editors have plugins that will
+apply `black` as you edit files.
+
+Optionally, you may wish to setup [pre-commit hooks](https://pre-commit.com)
+to automatically run `black` and ``flake8`` when you make a git commit. This
+can be done by installing `pre-commit` from the repository root:
+
+```
+pip install pre-commit
```
Comment (martindurant) [inline README.md]: (also on conda)
Note that you mention flake8 but don't include it in the instructions.

**Comment by asford** [comment]
Thanks for the catch on review, I've tidied up the documentation and travis build.

**Comment by martindurant** [comment]
I'd like to wait for #132 first, I hope that's OK.

**Comment by asford** [comment]
Of course, we should wait to merge if you've pending work. I'm happy to rebase against master whenever you're clear. As a note, here is the rebase sequence:

```bash
# remotes: asford, intake
git fetch intake
git fetch asford
git checkout -B black asford/black~1 
git rebase intake/master
pre-commit install --install-hooks
pre-commit run --all
git add .
git commit -m "Blacken."
git push --force-with-lease -u asford black
```

**Comment by martindurant** [comment]
I think you are clean to rebase and fix now.

**Comment by asford** [comment]
Rebased and ready to merge.

**Comment by martindurant** [comment]
Thank you!

**Pull request merged**